src/governance/governance.service.ts

```diff
@@ -0,0 +1,203 @@
+import { BlocksRangeOptions } from '../common';
+import { Network } from '../config';
+import { TimelockController, TimelockController__factory } from '../contracts';
+import {
+  CallDisputedEvent,
+  CallDisputedEventFilter,
+  CallDisputedResolvedEvent,
+  CallDisputedResolvedEventFilter,
+  CallExecutedEvent,
+  CallExecutedEventFilter,
+  CallScheduledEvent,
+  CallScheduledEventFilter,
+  CancelledEvent,
+  CancelledEventFilter,
+} from '../contracts/TimelockController';
+import { Service } from '../service';
+import { chunkQueryFilter } from '../utils';
+import { getBlockDeployedAt } from '../utils/deployed-at.util';
+
+export class GovernanceService extends Service {
+  static readonly TIMELOCK_CONTRACT_ADDR_MAP = {
+    [`${Network.Arbitrum}`]: '0xC4611233328559Dd1849947bC1e2b3B6b8714fa9',
+  };
+
+  private loading?: Promise<void>;
+  private _timelockController?: TimelockController;
+
+  async ready() {
+    if (!this.loading) {
+      this.loading = this.#init();
+    }
+
+    return this.loading;
+  }
+
+  get timelockAddress() {
+    return GovernanceService.TIMELOCK_CONTRACT_ADDR_MAP[
+      this.sdk.config.network
+    ];
+  }
+
+  get timelockController(): TimelockController {
+    if (!this._timelockController) {
+      throw new Error(
+        `Timelock controller is not defined for ${this.config.network}`,
+      );
+    }
+    return this._timelockController;
+  }
+
+  async loadScheduledProposals<T extends Required<BlocksRangeOptions>>(
+    { startBlock = 0, endBlock = 0 }: T = {
+      startBlock: 0,
+      endBlock: 0,
+    } as T,
+  ) {
+    const options = { startBlock, endBlock };
+    await this.#processEventsScanRange(this.timelockAddress, options);
+
+    const scheduledProposalsFilter =
+      this.timelockController.filters.CallScheduled();
+
+    return chunkQueryFilter<
+      TimelockController,
+      CallScheduledEventFilter,
+      CallScheduledEvent
+    >(
+      this.timelockController,
+      scheduledProposalsFilter,
+      options.startBlock,
+      options.endBlock,
+    );
+  }
+
+  async loadProposalsStatusChange<T extends Required<BlocksRangeOptions>>(
+    { startBlock = 0, endBlock = 0 }: T = {
+      startBlock: 0,
+      endBlock: 0,
+    } as T,
+  ) {
+    const options = { startBlock, endBlock };
+    await this.#processEventsScanRange(this.timelockAddress, options);
+
+    const executedProposalsFilter =
+      this.timelockController.filters.CallExecuted();
+    const canceledProposalsFilter = this.timelockController.filters.Cancelled();
```

Comment:
is canceled the same as veto or should we include a veto check here if available?

Reply:
veto = callDispute()